File: Data Structures & Algorithms/baseball-game/submission-1.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> record = new Stack<>();
        for(String op : operations) {
            if (op.equals("+")) {
                int a = record.pop();
                int b = record.pop();
                record.push(b);
                record.push(a);
                record.push(b+a);
            } else if (op.equals("C")) {
                record.pop();
            } else if (op.equals("D")) {
                record.push(record.peek() * 2);
            } else {
                int a = Integer.parseInt(op);
                record.push(a);
            }
        }
        int sum = 0;
        for (int a : record) {
            sum += a;
        }
        return sum;
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        sum = 0

        while sum != 1:
            sum = 0
            if n in seen:
                return False
            seen.add(n)
            while n > 0:
                a = n % 10
                n = n // 10
                sum += a**2
            n = sum
        return True